stor/dx.py

```diff
@@ -385,129 +1067,161 @@ def exists(self):
             # first see if there is a specific corresponding object
             self.stat()
             return True
-        except (NotFoundError, ValueError):
+        except (stor_exceptions.NotFoundError, ValueError):
             pass
-        # otherwise we could be a directory, so try to grab first
-        # file/subfolder
-        if self._is_folder():
-            try:
-                self.list(limit=1)
-                return True
-            except NotFoundError:
-                return False
-        return False
+        # otherwise we could be a directory, so try to listdir folder
+        # note list doesn't error on non-existent folder and cannot be used here
+        try:
+            self.listdir()
+            return True
+        except stor_exceptions.NotFoundError:
+            return False
 
+    def getsize(self):
+        if not self.resource:
+            return self.stat()['dataUsage']*1e9
+        else:
+            return self.stat()['size']
+
+    @_propagate_dx_exceptions()
     def stat(self):
-        if not self.__stat:
+        """Performs a stat on the path.
 
-            if self.canonical_resource:
-                self.__stat = dxpy.DXFile(dxid=self.canonical_resource,
-                                          project=self.canonical_project).describe()
-            else:
-                self.__stat = dxpy.DXProject(dxid=self.canonical_project).describe()
-        return self.__stat
+        Raises:
+            DuplicateError: If project or resource is not unique
+            NotFoundError: When the project or resource cannot be found
+            ValueError: If path is folder path
+        """
+        if not self.resource:
+            return dxpy.DXProject(dxid=self.canonical_project).describe()
+        return dxpy.DXFile(dxid=self.canonical_resource,
+                           project=self.canonical_project).describe()
 
 
 class DXVirtualPath(DXPath):
-    """Class Handler for DXPath of form 'dx://project-{ID}:/a/b/c' or 'dx://a/b/c'"""
+    """Class Handler for DXPath of form 'dx://project-{dxID}:/a/b/c' or 'dx://a:/b/c'"""
```

Comment:
for clarity can you reverse this:

```suggestion
    """Class Handler for DXPath of form 'dx://MyProject:/a/b/c' or 'dx://project-123:/b/c'"""
```

Reply:
additionally, to me `dxid` *includes* the type prefix

Reply:
Reversed, changed dxID to uuid as suggested.